public/js/cin.js

```diff
@@ -0,0 +1,527 @@
+        var intDate = new Date(interviews[i].start_time);
+        if (
+          intDate.getTime() > Date.now() &&
+          intDate.getTime() < Date.now() + 900000
+        ) {
+          setShowNext(true);
+          setPositions(["", interviews[i].interview.position.full_name]);
+          setCandidates([
+            "",
+            prettifyCandidates(interviews[i].interview.candidates),
+          ]);
+          setInterviewers(["", getInterviewers(interviews[i])]);
+          setTimes([
+            "",
+            intDate.toLocaleTimeString().slice(0, -3) +
+              " - " +
+              new Date(interviews[i].end_time)
+                .toLocaleTimeString()
+                .slice(0, -3),
+          ]);
+          break;
+        } else {
+          setShowNext(false);
+        }
+      }
+    }
+  };
+
+  useEffect(() => {
+    setTimeout(() => {
+      updateLives();
+    }, refreshTime);
+  });
+
+  return html`
+    <div class="row">
+      <${LiveCard}
+        class="col"
+        live="Live Now"
+        show=${showLive}
+        position=${positions[0]}
+        candidate=${candidates[0]}
+        interviewer=${interviewers[0]}
+        time=${times[0]}
+      />
+
+      <${LiveCard}
+        class="col"
+        live="Next Up"
+        show=${showNext}
+        position=${positions[1]}
+        candidate=${candidates[1]}
+        interviewer=${interviewers[1]}
+        time=${times[1]}
+      />
+    </div>
+  `;
+};
+
+const quickSortTime = (data) => {
+  if (data.length === 0) {
+    return [];
+  } else {
+    var before = [];
+    var after = [];
+    var pivot = data[0];
+    for (let i = 1; i < data.length; i++) {
+      if (
+        new Date(data[i].start_time).getTime() <
+        new Date(pivot.start_time).getTime()
+      ) {
+        before.push(data[i]);
+      } else {
+        after.push(data[i]);
+      }
+    }
+    return quickSortTime(before).concat([pivot]).concat(quickSortTime(after));
+  }
+};
+
+const getData = async () => {
+  console.log("Updating API Data");
+  fetch(api + "/events/")
+    .then((r) => r.json())
+    .then((data) => {
+      var usableEvents = [];
+      data.data.forEach((x) => {
+        if (x.interview != null) {
+          usableEvents.push(x);
+        }
+      });
+      interviews = quickSortTime(usableEvents);
+    });
+};
+
+const App = () => {
+  useEffect(() => {
+    setTimeout(() => {
+      getData();
+    }, refreshTime);
+  });
+
+  return html`
+    <${LiveArea} />
+    <${ScheduleArea} />
+  `;
+};
+
+const getApi = async () => {
+  const apiCall = await fetch("/cinapi");
+  const res = await apiCall.text();
+  return res;
+};
+
+getApi().then((x) => {
+  api = x;
+  getData().then(() => {
+    render(html`<${App} />`, interactive);
+
+    // This stuff is fun. It (mostly) works though.
+    setInterval(() => {
+      getData();
+    }, longTermRefreshTime);
+    setTimeout(() => (refreshTime = longTermRefreshTime), refreshTime * 10);
```

Comment:
Excuse me, what the fuck

Reply:
oh yeah, curly braces. good spot. the fun here is it needed a low refresh time to shove the data in first, otherwise it'd take 30 seconds to load the original data. there may be a better way to do it, but this fun worked, and also considered that it may not load the api data immediately (cause it won't)